From 501bb68a66cfc0bc2a2458483400cb49daca974f Mon Sep 17 00:00:00 2001
Message-Id: <501bb68a66cfc0bc2a2458483400cb49daca974f.1656310634.git.geliang.tang@suse.com>
From: "David E. Box" <david.e.box@linux.intel.com>
Date: Tue, 16 Mar 2021 19:44:54 -0700
Subject: [PATCH] platform/x86: intel_pmt_class: Initial resource to 0
Git-commit: 501bb68a66cfc0bc2a2458483400cb49daca974f
Patch-mainline: v5.13-rc1
References: bsn#171
Modified-by-SEL: No

Initialize the struct resource in intel_pmt_dev_register to zero to avoid a
fault should the char *name field be non-zero.

Signed-off-by: David E. Box <david.e.box@linux.intel.com>
Link: https://lore.kernel.org/r/20210317024455.3071477-1-david.e.box@linux.intel.com
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/platform/x86/intel_pmt_class.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel_pmt_class.c b/drivers/platform/x86/intel_pmt_class.c
index 228e21f1ce5c..c86ff15b1ed5 100644
--- a/drivers/platform/x86/intel_pmt_class.c
+++ b/drivers/platform/x86/intel_pmt_class.c
@@ -219,7 +219,7 @@ static int intel_pmt_dev_register(struct intel_pmt_entry *entry,
 				  struct intel_pmt_namespace *ns,
 				  struct device *parent)
 {
-	struct resource res;
+	struct resource res = {0};
 	struct device *dev;
 	int ret;
 
-- 
2.35.3

